### ISSUE TITLE
can not run `npx react-native init MyApp --template react-native-template-typescript`

### ISSUE THREAD

**Issue by Kennytian** [react-native-community/react-native-template-typescript#80]
# Bug

## Environment info

React native info output:

```bash
 // paste it here
```

Library version: x.x.x

## Steps To Reproduce

1.
2.
...

Describe what you expected to happen:

1.
2.

## Reproducible sample code

```
yarn add v1.17.3
[1/4] 🔍  Resolving packages...
warning jest > jest-cli > jest-config > jest-environment-jsdom > jsdom > left-pad@1.3.0: use String.prototype.padStart()
error An unexpected error occurred: "https://registry.npmjs.org/react-native-template-react-native-template-typescript: Not found".
info If you think this is a bug, please open a bug report with the information provided in "/Users/kenny/react0611/yarn-error.log".
info Visit https://yarnpkg.com/en/docs/cli/add for documentation about this command.
yarn remove v1.17.3
[1/2] 🗑  Removing module react-native-template-react-native-template-typescript...
error This module isn't specified in a package.json file.
info Visit https://yarnpkg.com/en/docs/cli/remove for documentation about this command.
warn Failed to clean up template temp files in node_modules/react-native-template-react-native-template-typescript. This is not a critical error, you can work on your app.
(node:2599) UnhandledPromiseRejectionWarning: Error: Command failed: yarn add react-native-template-react-native-template-typescript
    at makeError (/Users/kenny/react0611/node_modules/execa/index.js:174:9)
    at Promise.all.then.arr (/Users/kenny/react0611/node_modules/execa/index.js:278:16)
    at process._tickCallback (internal/process/next_tick.js:68:7)
(node:2599) UnhandledPromiseRejectionWarning: Unhandled promise rejection. This error originated either by throwing inside of an async function without a catch block, or by rejecting a promise which was not handled with .catch(). (rejection id: 3)
(node:2599) [DEP0018] DeprecationWarning: Unhandled promise rejections are deprecated. In the future, promise rejections that are not handled will terminate the Node.js process with a non-zero exit code.
k
```

**Comment by KingAmo** [comment]
same problem here
```
info Fetching template react-native-template-react-native-template-typescript...
yarn add v1.16.0
[1/4] 🔍  Resolving packages...
warning jest > jest-cli > jest-config > jest-environment-jsdom > jsdom > left-pad@1.3.0: use 
String.prototype.padStart()
error An unexpected error occurred: "https://registry.npmjs.org/react-native-template-react-
native-template-typescript: Not found".
info If you think this is a bug, please open a bug report with the information provided in 
"/Users/qinshuai/Documents/ReactNativeProj/RN_0.61/TsSimpleApp/yarn-error.log".
info Visit https://yarnpkg.com/en/docs/cli/add for documentation about this command.
yarn remove v1.16.0
[1/2] 🗑  Removing module react-native-template-react-native-template-typescript...
error This module isn't specified in a package.json file.
info Visit https://yarnpkg.com/en/docs/cli/remove for documentation about this command.
```

**Comment by PeterKow** [comment]
Just remember to `npm uninstall -g react-native-cli` and `yarn global add @react-native-community/cli` so you have new features during starting a new project

**Comment by radko93** [comment]
@Kennytian @KingAmo seems like you are still using old CLI. Please try as @PeterKow suggeseted

**Comment by Kennytian** [comment]
@PeterKow @radko93  Thanks for your response, following is my operate steps

Step 1:
```
kenny@Kennys-Mac-mini ~ $ npm uninstall -g react-native-cli
up to date in 0.035s
```

Step 2:
```
kenny@Kennys-Mac-mini ~ $ yarn global add @react-native-community/cli
yarn global v1.17.3
[1/4] 🔍  Resolving packages...
[2/4] 🚚  Fetching packages...
[3/4] 🔗  Linking dependencies...
warning "@react-native-community/cli > metro-react-native-babel-transformer@0.54.1" has unmet peer dependency "@babel/core@*".
warning " > @react-native-community/cli@2.9.0" has unmet peer dependency "react-native@^0.60.0".
[4/4] 🔨  Building fresh packages...

warning Your current version of Yarn is out of date. The latest version is "1.19.0", while you're on "1.17.3".
info To upgrade, run the following command:
$ brew upgrade yarn
success Installed "@react-native-community/cli@2.9.0" with binaries:
      - react-native
✨  Done in 8.83s.
```

Step 3:
```
yarn add v1.17.3
[1/4] 🔍  Resolving packages...
error An unexpected error occurred: "https://registry.npmjs.org/react-native-template-react-native-template-typescript: Not found".
info If you think this is a bug, please open a bug report with the information provided in "/Users/kenny/projects/reactnative/react0611/yarn-error.log".
info Visit https://yarnpkg.com/en/docs/cli/add for documentation about this command.
yarn remove v1.17.3
[1/2] 🗑  Removing module react-native-template-react-native-template-typescript...
error This module isn't specified in a package.json file.
info Visit https://yarnpkg.com/en/docs/cli/remove for documentation about this command.
warn Failed to clean up template temp files in node_modules/react-native-template-react-native-template-typescript. This is not a critical error, you can work on your app.
(node:619) UnhandledPromiseRejectionWarning: Error: Command failed: yarn add react-native-template-react-native-template-typescript
```

The same issue here :(

**Comment by KingAmo** [comment]
@Kennytian maybe you also need run ` yarn global remove react-native-cli`

**Comment by Kennytian** [comment]
```
               ######                ######               
             ###     ####        ####     ###             
            ##          ###    ###          ##            
            ##             ####             ##            
            ##             ####             ##            
            ##           ##    ##           ##            
            ##         ###      ###         ##            
             ##  ########################  ##             
          ######    ###            ###    ######          
      ###     ##    ##              ##    ##     ###      
   ###         ## ###      ####      ### ##         ###   
  ##           ####      ########      ####           ##  
 ##             ###     ##########     ###             ## 
  ##           ####      ########      ####           ##  
   ###         ## ###      ####      ### ##         ###   
      ###     ##    ##              ##    ##     ###      
          ######    ###            ###    ######          
             ##  ########################  ##             
            ##         ###      ###         ##            
            ##           ##    ##           ##            
            ##             ####             ##            
            ##             ####             ##            
            ##          ###    ###          ##            
             ###     ####        ####     ###             
               ######                ######               
                                                          

                  Welcome to React Native!                
                 Learn Once Write Anywhere                

✔ Downloading template
✖ Copying template
error RangeError: Maximum call stack size exceeded
```

Show some different issue

**Comment by radko93** [comment]
@Kennytian what is Step 3?

**Comment by PeterKow** [comment]
Confirm, I've run in the same issue in the end `✖ Copying template
error RangeError: Maximum call stack size exceeded`

Then just end up installing RN and adding TS manually as in here -> https://facebook.github.io/react-native/docs/typescript

Sorry for late reply

**Comment by Kennytian** [comment]
> @Kennytian what is Step 3?

npx react-native init react0611 --template react-native-template-typescript
This will walk you through creating a new React Native project in /Users/kenny/projects/private/reactnative/react0611
Using yarn v1.17.3
Installing react-native...
yarn add v1.17.3

**Comment by MateusAndrade** [comment]
Just tested it here without any global package related to react-native cli on the global packages and it returns a similar error: 

```
Removing module react-native-template-react-native-template-typescript...
error This module isn't specified in a package.json file.
info Visit https://yarnpkg.com/en/docs/cli/remove for documentation about this command.
warn Failed to clean up template temp files in node_modules/react-native-template-react-native-template-typescript. This is not a critical error, you can work on your app.
```

Tested with: ```npx react-native init MyApp --template react-native-template-typescript``` and ```npx react-native init MyApp --template react-native-template-typescript@6.2.0```

**Comment by emin93** [comment]
The error message looks very much like it's coming from the old CLI because it's prepending `react-native-template-` to the package name.

I tried the following scenarios (macOS 10.14.6):
- Global package `@react-native-community/cli` installed, no `yarn` installed
- Global package `@react-native-community/cli` installed, `yarn` installed
- No global CLI installed, `yarn` installed
- No global CLI installed, no `yarn` installed

Everything seemed to work just fine, so I'm kind of running out of ideas.

**Comment by emin93** [comment]
Can you try adding the `--ignore-existing` flag?

`npx --ignore-existing react-native init MyApp --template react-native-template-typescript`

**Comment by Kennytian** [comment]
@emin93  Thanks for your reply, `npx --ignore-existing react-native init MyApp --template react-native-template-typescript` it's work.

Thanks all the same @KingAmo @PeterKow @radko93 @MateusAndrade !

**Comment by GreatAuk** [comment]
> Just remember to `npm uninstall -g react-native-cli` and `yarn global add @react-native-community/cli` so you have new features during starting a new project

I have a project with RN@0.58.6 and legacy `react-native-cli`， If I do this will affect my previous project ?

**Comment by radko93** [comment]
@GreatAuk you can always install the old CLI again if anything bad happens. But I think it should be fine.

**Comment by GreatAuk** [comment]
> @GreatAuk you can always install the old CLI again if anything bad happens. But I think it should be fine.

`npm install -g react-native-cli` still download old CLI, not https://github.com/react-native-community/cli ?

**Comment by radko93** [comment]
@GreatAuk yeah you can check on npm https://www.npmjs.com/package/react-native-cli

**Comment by tgmarinho** [comment]
@emin93 your solution works for me! Thanks!

[https://www.tgmarinho.com/how_to_install_react_native_with_typescript_without_react_native_cli/](https://www.tgmarinho.com/how_to_install_react_native_with_typescript_without_react_native_cli/)

**Comment by zaguiini** [comment]
Guys, this error happened to me and I don't have the CLI globally installed (ran from `npx`)...

**Comment by Domino987** [comment]
@emin93 `npx --ignore-existing react-native init MyApp --template react-native-template-typescript `does unfortunately not work for me. I uninstalled the react-native-cl and tried several different way to execute the npx call. Is there another way to get a new project set up with this?

**Comment by radko93** [comment]
@Domino987 you are still somehow using the old CI if it fails. You can try installing new CLI globally but that's not recommended

**Comment by tsirolnik** [comment]
Running the following command - 
```
npx react-native init app --template react-native-template-typescript@6.2.0
```
Which is being introduced in the official React Native website [here](https://facebook.github.io/react-native/docs/typescript) (substituted MyTSProject with app)

Results in the same output as above.
```
error An unexpected error occurred: "https://registry.yarnpkg.com/react-native-template-react-native-template-typescript: Not found".
info If you think this is a bug, please open a bug report with the information provided in "/home/user/dev/App/mobile/app/yarn-error.log".
info Visit https://yarnpkg.com/en/docs/cli/add for documentation about this command.
yarn remove v1.13.0
[1/2] Removing module react-native-template-react-native-template-typescript@6.2.0...
error This module isn't specified in a package.json file.
info Visit https://yarnpkg.com/en/docs/cli/remove for documentation about this command.
warn Failed to clean up template temp files in node_modules/react-native-template-react-native-template-typescript@6.2.0. This is not a critical error, you can work on your app.
(node:31167) UnhandledPromiseRejectionWarning: Error: Command failed: yarn add react-native-template-react-native-template-typescript@6.2.0
    at makeError (/home/user/dev/App/mobile/app/node_modules/execa/index.js:174:9)
    at Promise.all.then.arr (/home/user/dev/App/mobile/app/node_modules/execa/index.js:278:16)
    at process._tickCallback (internal/process/next_tick.js:68:7)
(node:31167) UnhandledPromiseRejectionWarning: Unhandled promise rejection. This error originated either by throwing inside of an async function without a catch block, or by rejecting a promise which was not handled with .catch(). (rejection id: 3)
(node:31167) [DEP0018] DeprecationWarning: Unhandled promise rejections are deprecated. In the future, promise rejections that are not handled will terminate the Node.js process with a non-zero exit code.
```

**Comment by duongntb94** [comment]
After I upgraded to yarn version 1.21.1, this problem has been resolved.

**Comment by leotm** [comment]
My journey was from [getting-started](https://reactnative.dev/docs/getting-started) to https://github.com/react-native-community/react-native-template-typescript/issues/80#issuecomment-536175565 to https://github.com/react-native-community/react-native-template-typescript/issues/80#issuecomment-536419979 for a successful basic setup, glad we have this thread 😅

**Comment by zuigzm** [comment]
You can use `npx react-native init MyApp --template typescript`, it will run normally

**Comment by carlosstenzel** [comment]
## Best solution I found

Remove

`npm uninstall -g react-native-cli`

Create the folder where the project will be

`mkdir [folder name] && cd [folder name]`

Install the dependencies and everything will work

```
yarn init
yarn add react-native
yarn react-native init [Project Name] ...
```

**Comment by faagner7** [comment]
> Can you try adding the `--ignore-existing` flag?
> 
> `npx --ignore-existing react-native init MyApp --template react-native-template-typescript`

thanks for your reply! It worked for me!

**Comment by pedroigor66** [comment]
> ## Best solution I found
> Remove
> 
> `npm uninstall -g react-native-cli`
> 
> Create the folder where the project will be
> 
> `mkdir [folder name] && cd [folder name]`
> 
> Install the dependencies and everything will work
> 
> ```
> yarn init
> yarn add react-native
> yarn react-native init [Project Name] ...
> ```

Thank you so much, @carlosstenzel. Your solution was the only one that worked for me.

**Comment by uranium93** [comment]
> Can you try adding the --ignore-existing flag?
> 
> npx --ignore-existing react-native init MyApp --template react-native-template-typescript

By @emin93  **worked perfectly**, seems the issue with npm | yarn cache

**Comment by mk1020** [comment]
Try update yarn.
Run curl --compressed -o- -L https://yarnpkg.com/install.sh | bash
It helped me

**Comment by methineeinkan** [comment]
> ## Best solution I found
> Remove
> 
> `npm uninstall -g react-native-cli`
> 
> Create the folder where the project will be
> 
> `mkdir [folder name] && cd [folder name]`
> 
> Install the dependencies and everything will work
> 
> ```
> yarn init
> yarn add react-native
> yarn react-native init [Project Name] ...
> ```

work for me

**Comment by antoci-alin** [comment]
Guys? Is there any solving for this issue or should we move to Flutter?
I have like 1 week of trying to initiate a new RN project but I still got the downloading template issue. Sorry for being sarcastic but it's just frustrating to loose time trying to fix such an ordinary issue.

**Comment by YogiHa** [comment]
After uninstalling the old cli

```
npm uninstall -g react-native-cli
```

and install the new one

```
npm i -g @react-native-community/cli
```
and upgrade node

```
npm install -g npm@latest
```

the template created smoothly

**Comment by AbanoubAshraaf** [comment]
This template only works with the new CLI. Make sure you have uninstalled the legacy react-native-cli first (npm uninstall -g react-native-cli) for the below command to work. If you wish to not use npx, you can also install the new CLI globally (npm i -g @react-native-community/cli or yarn global add @react-native-community/cli).

If you tried the above and still get the react-native-template-react- native-template-typescript: Not found error, please try adding the --ignore-existing flag to force npx to ignore any locally installed versions of the CLI and use the latest.

Further information can be found here: https://github.com/react-native-community/cli#about

**Comment by AmorimRob** [comment]
In my case after uninstalling the CLI, I also had to update the node version to 16.14.0 to work properly

**Comment by asadMumtaz-YSTL** [comment]
Hi friends,

I'm facing this error kindly guide me how I resolve this issue.

I use command for creating new project "npx react-native init project_name"


× Downloading template
error Error: Command failed: yarn add react-native@latest
warning react-native > @react-native-community/cli > @react-native-community/cli-plugin-metro > metro > metro-minify-uglify > uglify-es@3.3.9: support for ECMAScript is superseded by `uglify-js` as of v3.13.0
warning react-native > react-native-codegen > jscodeshift > micromatch > snapdragon > source-map-resolve@0.5.3: See https://github.com/lydell/source-map-resolve#deprecated
warning react-native > react-native-codegen > jscodeshift > micromatch > snapdragon > source-map-resolve > source-map-url@0.4.1: See https://github.com/lydell/source-map-url#deprecated
warning react-native > react-native-codegen > jscodeshift > micromatch > snapdragon > source-map-resolve > resolve-url@0.2.1: https://github.com/lydell/resolve-url#deprecated
warning react-native > react-native-codegen > jscodeshift > micromatch > snapdragon > source-map-resolve > urix@0.1.0: Please see https://github.com/lydell/urix#deprecated
error react-native@0.68.1: The engine "node" is incompatible with this module. Expected version ">=14". Got "12.13.0"
error Found incompatible module.
yarn add v1.22.17
info No lockfile found.
[1/4] Resolving packages...
[2/4] Fetching packages...
info Visit https://yarnpkg.com/en/docs/cli/add for documentation about this command.

**Comment by cliffgor** [comment]
in my case I had to reset yarn `yarn set version 1.22.1` my yarn version was `3.2.0`

**Comment by shan-github** [comment]
This works for me 🎉

`npx react-native init ProjectName --npm`

**Comment by Arunshaik2001** [comment]
> ## Best solution I found
> Remove
> 
> `npm uninstall -g react-native-cli`
> 
> Create the folder where the project will be
> 
> `mkdir [folder name] && cd [folder name]`
> 
> Install the dependencies and everything will work
> 
> ```
> yarn init
> yarn add react-native
> yarn react-native init [Project Name] ...
> ```

This worked for me Thank You

**Comment by davuttemel** [comment]
npx react-native@latest init AwesomeProject --pm npm
